pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -100,14 +100,21 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 
 	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
-		instanceOptedIn, err := r.poolManager.IsVmInstanceOptedIn(request.Namespace)
+		vmOptMode, err := r.poolManager.GetVmOptMode()
 		if err != nil {
-			return reconcile.Result{}, errors.Wrap(err, "failed to check opt-in selection for vm")
+			return reconcile.Result{}, errors.Wrap(err, "failed to get opt-Mode")
 		}
+		if vmOptMode == pool_manager.OptInMode {
+			instanceOptedIn, err := r.poolManager.IsVmInstanceOptedIn(request.Namespace)
 
-		if !instanceOptedIn {
-			logger.Info("vm is opted-out from kubemacpool")
-			return reconcile.Result{}, nil
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to check opt-in selection for vm")
+			}
+
+			if !instanceOptedIn {
+				logger.Info("vm is opted-out from kubemacpool")
+				return reconcile.Result{}, nil
+			}
```

Comment:
Hey could you please elaborate on that, how is the behavior is changed?

Reply:
`IsVmInstanceOptedIn()` is currently used to exit the controller if the vm is not "opted-in", i.e. the namespace of the vm doesn't have the right label.

In Opt-out mode however, we want to opposite behavior, i.e. that the controller will **not** exit by default.

This basically means that `IsVmInstanceOptedIn()` needs to only be checked if we are indeed in `opt-in mode`. So I added a check for that